app/views/layouts/menus/_student_sub_menu.html.erb

```diff
@@ -65,7 +65,7 @@
         </li>
       <% end %>
 
-      <% if @assignment.has_peer_review_assignment? && !@assignment.pr_assignment.is_hidden %>
+      <% if @assignment.has_peer_review_assignment? && allowed_to?(:see_hidden?, @assignment.pr_assignment.id) %>
```

Comment:
Remove the `.id` at the end. Otherwise the `record` will be the assignment's id (an integer) not the assignment object itself.